src/services/QueryService.ts

```diff
@@ -96,23 +96,57 @@ export class QueryService<T> {
   async get(add): Promise<BullhornListResponse<T>> {
     return this.run(add);
   }
-  async run(add): Promise<BullhornListResponse<T>> {
+  async run(add: boolean): Promise<BullhornListResponse<T>> {
     await this.initialized;
-    return Promise.all([this.http.get(this.endpoint, { params: this.parameters }), this.meta.getFull(this.parameters.fields, this.parameters.layout)])
-      .then(([response, metadata]) => [response.data, metadata])
-      .then(([result, metadata]: [BullhornListResponse<T>, BullhornMetaResponse]) => {
-        this._lastResponse = result;
-        const records = result.data;
-        if (add) {
-          this.records = this.records.concat(records);
-        } else {
-          this.records = records;
-        }
-        result.meta = metadata;
-        return result;
-      });
-  }
-  async then(done: any, fail?: any): Promise<any> {
+    const [response, metadata] = await Promise.all([this.httpGet(this.parameters), this.meta.getFull(this.parameters.fields, this.parameters.layout)]);
+    const result = response.data;
+    if (this.shouldPullMoreRecords(result)) {
+      const recursiveData = await this.recursiveQueryPull(result);
+      result.data = result.data.concat(recursiveData);
+    }
+    result.count = result.data.length;
+
+    console.log(result);
+    this._lastResponse = result;
+    if (add) {
+      this.records = this.records.concat(result.data);
+    } else {
+      this.records = result.data;
+    }
+    result.meta = metadata;
+    return result;
+  }
+
+  private async recursiveQueryPull({ count = 0, data = [], start = 0, total = 0 }) {
+    const [nextStart, nextCount] = this.getNext(start, count);
+    const response = await this.httpGet({ ...this.parameters, ...{ start: nextStart, count: nextCount } });
+    if (this.shouldPullMoreRecords(response.data)) {
+      const nextData = await this.recursiveQueryPull(response.data);
+      response.data.data = response.data.data.concat(nextData);
+    }
+    return response.data.data;
+  }
+
+  private shouldPullMoreRecords({ count = 0, data = [], start = 0, total = 0 }) {
+    const [nextStart, nextCount] = this.getNext(start, count);
+    if (nextStart < total && count !== 0) {
+      return nextCount;
+    }
+    return 0;
+  }
+
+  private getNext(start: number, count: number) {
+    const nextStart = start + count;
+    const alreadyFetchRecord = nextStart - this.parameters.start;
+    const nextCount = this.MAX_RECORDS - alreadyFetchRecord;
```

Comment:
Super minor - consider changing variable name to: alreadyFetched

Reply:
done